### ISSUE TITLE
`simplify-generator` doesn't quite simplify enough

### ISSUE THREAD

**Issue by henryiii** [sourcery-ai/sourcery#124]
### Issue description or question

This doesn't quite get simplified enough:

https://github.com/henryiii/cibuildwheel/pull/1#discussion_r598904665

It does:

```diff
- binary_data = bytes(n for n in range(1, 256))
+ binary_data = bytes(iter(range(1, 256)))
```

When it would be better to do:

```diff
binary_data = bytes(n for n in range(1, 256))
binary_data = bytes(range(1, 256))
```

This [already takes an Iterable](https://github.com/python/typeshed/blob/29c3d67626b2103eb9ab1b88fdf1e6e30a6bbde2/stdlib/builtins.pyi#L412), it doesn't need to be an `Iterator`.

### Sourcery Version

Online (local too).

**Comment by Hellebore** [comment]
Thanks for pointing this out - will look into it.

**Comment by Hellebore** [comment]
Fixed in 0.9.1.
